src/gluonts/model/deeprenewal/_network.py

```diff
@@ -0,0 +1,629 @@
+    @validated()
+    def __init__(
+        self,
+        num_layers: int,
+        num_cells: int,
+        cell_type: str,
+        history_length: int,
+        context_length: int,
+        prediction_length: int,
+        distr_output_m: DistributionOutput,
+        distr_output_q: DistributionOutput,
+        dropout_rate: float,
+        cardinality: List[int],
+        embedding_dimension: List[int],
+        lags_seq: List[int],
+        scaling: bool = True,
+        dtype: DType = np.float32,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.num_layers = num_layers
+        self.num_cells = num_cells
+        self.cell_type = cell_type
+        self.history_length = history_length
+        self.context_length = context_length
+        self.prediction_length = prediction_length
+        self.dropout_rate = dropout_rate
+        self.cardinality = cardinality
+        self.embedding_dimension = embedding_dimension
+        self.num_cat = len(cardinality)
+        self.scaling = scaling
+        self.dtype = dtype
+
+        assert len(cardinality) == len(
+            embedding_dimension
+        ), "embedding_dimension should be a list with the same size as cardinality"
+
+        assert len(set(lags_seq)) == len(lags_seq), "no duplicated lags allowed!"
+        lags_seq.sort()
+
+        self.lags_seq = lags_seq
+        # 2 separate distributions
+        self.distr_output_m = distr_output_m
+        self.distr_output_q = distr_output_q
+        RnnCell = {"lstm": mx.gluon.rnn.LSTMCell, "gru": mx.gluon.rnn.GRUCell}[
+            self.cell_type
+        ]
+
+        self.target_shape = distr_output_m.event_shape
+
+        # TODO: is the following restriction needed?
+        assert (
+            len(self.target_shape) <= 1
+        ), "Argument `target_shape` should be a tuple with 1 element at most"
+
+        with self.name_scope():
+            self.proj_distr_args_m = self.distr_output_m.get_args_proj(prefix='m')
+            self.proj_distr_args_q = self.distr_output_q.get_args_proj(prefix='q')
+            self.rnn = mx.gluon.rnn.HybridSequentialRNNCell()
+            for k in range(num_layers):
+                cell = RnnCell(hidden_size=num_cells)
+                cell = mx.gluon.rnn.ResidualCell(cell) if k > 0 else cell
+                cell = (
+                    mx.gluon.rnn.ZoneoutCell(cell, zoneout_states=dropout_rate)
+                    if dropout_rate > 0.0
+                    else cell
+                )
+                self.rnn.add(cell)
+            self.rnn.cast(dtype=dtype)
+            self.embedder = FeatureEmbedder(
+                cardinalities=cardinality,
+                embedding_dims=embedding_dimension,
+                dtype=self.dtype,
+            )
+            if scaling:
+                self.scaler = MeanScaler(keepdims=True)
+            else:
+                self.scaler = NOPScaler(keepdims=True)
+
+    @staticmethod
+    def get_lagged_subsequences(
```

Comment:
I understand both this method and `unroll_encoder` are from DeepARNetwork. Can we reuse DeepARNetwork's code without copy-pasting it? Maybe via overriding it?